pkg/config/initoptions.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+)
+
+const manifestFilename = "grouping-object-template.yaml"
+
+const configMapTemplate = `# NOTE: auto-generated
+# Date: <DATETIME>
+#
+# Contains the "grouping object" template ConfigMap.
+# When this object is applied, it is handled specially,
+# storing the metadata of all the other objects applied.
+# This object and its stored inventory is subsequently
+# used to calculate the set of objects to automatically
+# delete (prune), when an object is omitted from further
+# applies.
+#
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: <NAMESPACE>
+  name: grouping-object
+  labels:
+    # This label defines the ConfigMap as a grouping object template
+    cli-utils.sigs.k8s.io/inventory-id: <GROUPNAME>
+`
+
+// InitOptions contains the fields necessary to generate a
+// grouping object template ConfigMap.
+type InitOptions struct {
+	ioStreams genericclioptions.IOStreams
+	// Package directory argument
+	Dir string
+	// Namespace for grouping object
+	Namespace string
+	// Grouping object label value
+	GroupName string
+	// Random seed
+	Seed int64
+}
+
+func NewInitOptions(ioStreams genericclioptions.IOStreams) *InitOptions {
+	return &InitOptions{
+		ioStreams: ioStreams,
+	}
+}
+
+// Complete fills in the InitOptions fields.
+func (i *InitOptions) Complete(args []string) error {
```

Comment:
I'm not sure how we will wire this up. But I don't think we should continue this pattern from Kubectl with the options struct and the Complete function. I think the way kpt commands work are cleaner.

Reply:
I'll add a TODO.